src/fu-device-list.c

```diff
@@ -723,7 +723,7 @@ fu_device_list_add (FuDeviceList *self, FuDevice *device)
 	if (item != NULL &&
 	    g_strcmp0 (fu_device_get_protocol (item->device),
 		       fu_device_get_protocol (device)) == 0) {
-		if (!fu_device_has_flag (device, FWUPD_DEVICE_FLAG_NO_GUID_MATCHING)) {
+		if (fu_device_has_internal_flag (device, FU_DEVICE_INTERNAL_FLAG_COLDPLUG_MATCH_GUID)) {
 			if (fu_device_get_priority (device) < fu_device_get_priority (item->device)) {
 				g_debug ("ignoring device %s [%s:%s] as better device %s [%s:%s] already exists",
 					 fu_device_get_id (device),
```

Comment:
is this logic actually right?

Reply:
Yup, it's doing the checks there based on GUIDs.